File: Teledyne DALSA/Sapera/Demos/NET/BayerDemo/VB/SapMyProcessing.vb
Public Class SapMyProcessing
   Inherits SapProcessing
   Private m_Bayer As SapBayer
   ' Constructor

   Public Sub New(ByVal pBuffers As SapBuffer, ByVal pBayer As SapBayer, ByVal pCallback As SapProcessingDoneHandler, ByVal pContext As Object)
      MyBase.New(pBuffers)
      MyBase.ProcessingDoneEnable = True
      AddHandler MyBase.ProcessingDone, pCallback
      MyBase.ProcessingDoneContext = pContext
      m_Bayer = pBayer
   End Sub

   Public Overloads Overrides Function Run() As Boolean
      If m_Bayer.Enabled AndAlso m_Bayer.SoftwareConversion Then
         m_Bayer.Convert(MyBase.Index)
      End If
      Return True
   End Function
End Class

File: Teledyne DALSA/Sapera/Demos/NET/BayerDemo/CSharp/SapMyProcessing.cs
using System;
using System.Collections.Generic;
using System.Text;
using DALSA.SaperaLT.SapClassBasic;

namespace DALSA.SaperaLT.Demos.NET.CSharp.BayerDemo
{
   class SapMyProcessing : SapProcessing   
   {
      private SapBayer m_Bayer;
     // Constructor

      public SapMyProcessing(SapBuffer pBuffers, SapBayer pBayer, SapProcessingDoneHandler pCallback, Object pContext)
        : base(pBuffers)
     {
        base.ProcessingDoneEnable = true;
        base.ProcessingDone += pCallback;
        base.ProcessingDoneContext = pContext;
        m_Bayer = pBayer;
     }

     public override bool Run()
     {
        if (m_Bayer.Enabled && m_Bayer.SoftwareConversion)
        {
           m_Bayer.Convert(base.Index);
        }

        return true;
     }
 
   }
}
